src/main/java/com/aws/greengrass/authorization/AuthorizationPolicyParser.java

```
@@ -132,23 +131,25 @@ private Map<String, List<AuthorizationPolicy>> parseAllPoliciesForComponent(Node
         */
         try {
             if (accessControlTopic instanceof Topics) {
-                accessControlMap = OBJECT_MAPPER.convertValue(((Topics)accessControlTopic).toPOJO(),
-                        new TypeReference<Map<String, Map<String, AuthorizationPolicyConfig>>>() {
+                accessControlMap = OBJECT_MAPPER.convertValue(((Topics) accessControlTopic)
+                        .toPOJO(), new TypeReference<Map<String, Map<String, AuthorizationPolicyConfig>>>() {
                         });
             } else if (accessControlTopic instanceof Topic) {
-                // If its a leaf node, try parsing it. Be flexible and do best to understand customer intent
-                accessControlMap = OBJECT_MAPPER.readValue(Coerce.toString(accessControlTopic),
-                        new TypeReference<Map<String, Map<String, AuthorizationPolicyConfig>>>() {
+                // If its a leaf node, try parsing it. Be flexible and do best to understand
+                // customer intent
+                accessControlMap = OBJECT_MAPPER.readValue(Coerce
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `ObjectMapper.readValue()` method can throw a `JsonProcessingException`, but this exception is not handled properly here. By simply logging the error and returning an empty value, the exception is ignored. This can mask potential JSON parsing issues, leading to unexpected application behavior or data corruption because the system may continue with invalid or incomplete data. To fix this, catch `JsonProcessingException` and either handle it appropriately or rethrow it as a custom runtime exception. [https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)